src/main/drivers/barometer/barometer_lps.c

```diff
@@ -0,0 +1,166 @@
+
+#include "build/build_config.h"
+
+#include "barometer.h"
+#include "barometer_lps.h"
+
+#include "drivers/bus_spi.h"
+#include "drivers/io.h"
+#include "drivers/time.h"
+
+//====================================Registers Addresses=========================================// 
+#define LPS25HB_REF_P_XL    0x08
+#define LPS25HB_REF_P_L     0x09
+#define LPS25HB_REF_P_H     0x0A
+#define LPS25HB_WHO_AM_I    0x0F
+#define LPS25HB_RES_CONF    0x10
+#define LPS25HB_CTRL1       0x20
+#define LPS25HB_CTRL2       0x21
+#define LPS25HB_CTRL3       0x22
+#define LPS25HB_CTRL4       0x23
+#define LPS25HB_INT_CFG     0x24
+#define LPS25HB_INT_SOURCE  0x25
+#define LPS25HB_STATUS      0x27
+#define LPS25HB_OUT_XL      0x28
+#define LPS25HB_OUT_L       0x29
+#define LPS25HB_OUT_H       0x2A
+#define LPS25HB_TEMP_OUT_L  0x2B
+#define LPS25HB_TEMP_OUT_H  0x2C
+#define LPS25HB_FIFO_CTRL   0x2E
+#define LPS25HB_FIFO_STATUS 0x2F
+#define LPS25HB_THS_PL      0x30
+#define LPS25HB_THS_PH      0x31
+#define LPS25HB_RPDS_L      0x39
+#define LPS25HB_RPDS_H      0x3A
+//=======================================Constants=============================================// 
+#define LPS25HB_ID          0xBD
+#define LPS25HB_READ        0x80
+#define LPS25HB_MULT        0x40
+//======================================ODR Values=============================================//
+// CTRL1 value
+#define LPS25HB_ODR_ONE_SHOT        0x00
+#define LPS25HB_ODR_1               0x10
+#define LPS25HB_ODR_7               0x20
+#define LPS25HB_ODR_12_5            0x30
+#define LPS25HB_ODR_25              0x40
+//======================================Average number=============================================//
+#define LPS25HB_AVP_8               0x00
+#define LPS25HB_AVP_32              0x01
+#define LPS25HB_AVP_128             0x02
+#define LPS25HB_AVP_512             0x03
+#define LPS25HB_AVT_8               0x00
+#define LPS25HB_AVT_16              0x04
+#define LPS25HB_AVT_32              0x08
+#define LPS25HB_AVT_64              0x0C
+//======================================Moving average number=============================================//
+#define LPS25HB_NO_MA               0x00
+#define LPS25HB_MA_2                0x01
+#define LPS25HB_MA_4                0x03
+#define LPS25HB_MA_8                0x07
+#define LPS25HB_MA_16               0x0F
+#define LPS25HB_MA_32               0x1F
+
+bool lpsWriteCommand(busDevice_t *busdev, uint8_t cmd, uint8_t byte) {
+    return spiBusWriteRegister(busdev, cmd, byte);
+}
+
+bool lpsReadCommand(busDevice_t *busdev, uint8_t cmd, uint8_t len, uint8_t *data) {
+    return spiBusReadRegisterBuffer(busdev, cmd | 0x80 | 0x40, data, len);
+}
+
+static void lpsOn(busDevice_t *busdev, uint8_t CTRL1_val) {
+    lpsWriteCommand(busdev, LPS25HB_CTRL1, CTRL1_val | 0x80);
+    delay(500);
+}
+
+static void lpsOff(busDevice_t *busdev) {
+    lpsWriteCommand(busdev, LPS25HB_CTRL1, 0x00 | (0x01 << 2));
+}
+
+uint32_t rawP = 0;
```

Comment:
both `rawP` and `rawT` are signed, `watT` is 16bit : 
```
int32_t rawP;   // read as 24bit, sign extended
int16_t rawT;

Reply:
correct, this error came up from the fact that normally when I use this code I cast uint32 to int32 and perform calculations straight away to obtain Pa result. Casting on temperature is obviously wrong, it's not used afaik, but I will fix it so we don't have rubbish mistakes like this

Reply:
I looked at the code again it's ok with uint32 I'm casting those variables later on so It's fine.

Reply:
Yes, the code is working. But it is IMO better to use correct type for variables. Casting on use only increases confusion ...

Reply:
The only reason I do it this way is because for me it’s clearer to read out always values from registers into unsigned variable. Same was also done in ST implementation... It's my coding style but if everyone wishes I can change this spot...

Reply:
In this case it's a bit complicated - `rawP` value is sign-extended, so it's not raw value from register.  

`rawT` is 16 bit, storing it in 32bit register (without sign-extension) is confusing. Processed (and signed, but see https://github.com/betaflight/betaflight/pull/4857#discussion_r158964922) value is stored with current code, its a bit strange to have four billion degrees outside ;-) ..